server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -237,12 +236,8 @@ private CreateIndexRequest mapping(String type, BytesReference source, XContentT
             throw new IllegalStateException("mappings for type \"" + type + "\" were already defined");
         }
         Objects.requireNonNull(xContentType);
-        try {
-            mappings.put(type, XContentHelper.convertToJson(source, false, false, xContentType));
-            return this;
-        } catch (IOException e) {
-            throw new UncheckedIOException("failed to convert to json", e);
-        }
+        Map<String, Object> mappingAsMap = XContentHelper.convertToMap(source, false, xContentType).v2();
```

Comment:
> Also I think we can remove if (mappings.containsKey(type)) { ... } above

You're right, done.

Reply:
Yeah, I think parsing the mapping a second time is probably ok.  Would be nice to avoid it obviously, but I don't think it'll have a big impact since create index API calls should be relatively infrequent. 

++ to removing the `if (mappings.containsKey())` check, looks good to me